### PR TITLE
Ensure including a module doesn't prevent instance_methods from working

### PR DESCRIPTION
**Pull request by wied03**
Opal 0.9 had this case stop working

### CODE REVIEW

File: core/module/instance_methods_spec.rb [hunk 1]
```diff
@@ -6,6 +6,11 @@
     methods = ModuleSpecs::Parent.instance_methods(false)
     methods.should_not include(:undefed_method)
   end
+  
+  it "does not consider methods as superclass methods if invoked on the parent module" do
```
Comment (eregon) [inline core/module/instance_methods_spec.rb]: I do not understand the description. Is it about instance_methods only returning the methods of the module?
But that module does not have a parent, right?

Comment (wied03) [inline core/module/instance_methods_spec.rb]: Opal was having an issue where when a module was included in a class, etc., you could no longer retrieve the instance methods on the base module. This is trying to ensure you can still do that.

Comment (eregon) [inline core/module/instance_methods_spec.rb]: One important spec is missing here: `instance_methods(include_super=false)` includes only methods defined on that module.
Maybe there is a way to combine that and test when a module is included that the class `instance_methods(false)` does not change, as well as the module's.

**Comment by wied03** [comment]
@eregon - Tried to incorporate your feedback

**Comment by eregon** [comment]
Nice spec!

**Pull request merged**